src/components/ec/cuda/ec_cuda_executor_interruptible.c

```diff
@@ -0,0 +1,168 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2020-2022.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include "ec_cuda_executor.h"
+#include "components/mc/ucc_mc.h"
+#include "utils/ucc_atomic.h"
+
+ucc_status_t ucc_cuda_executor_interruptible_get_stream(cudaStream_t *stream)
+{
+    static uint32_t last_used   = 0;
+    int             num_streams = EC_CUDA_CONFIG->exec_num_streams;
+    cudaStream_t    tmp_stream;
+    cudaError_t     cuda_st;
+    int             i, j;
+    uint32_t        id;
+
+    if (ucc_unlikely(ucc_ec_cuda.exec_streams[num_streams - 1] == NULL)) {
+        ucc_spin_lock(&ucc_ec_cuda.init_spinlock);
+        if (ucc_ec_cuda.exec_streams[num_streams - 1] != NULL) {
+            goto unlock;
+        }
+
+        for(i = 0; i < num_streams; i++) {
+            cuda_st = cudaStreamCreateWithFlags(&tmp_stream,
+                                                cudaStreamNonBlocking);
+            ucs_atomic_cswap64((uint64_t*)(uintptr_t)&ucc_ec_cuda.exec_streams[i],
```

Comment:
@Sergei-Lebedev plz note we changed the fix in master. Per Devendars suggestion i've removed atomic cswap in favor of lobal ucc_ec_cuda.initialized in variable. This is done to make no assumption that cudaStram_t is 8 bytes (ptr). plz change accordingly

Reply:
done